McMasterNEUDOSE/cFS
# Adding LibCSP into CFS build

**Describe the contribution**
- Added the libcsp source code to `apps/libcsp`
- Created a script `run.sh` to build CSP and CFS
- CSP is built as a static library in the same build directory as the other CFS apps
- The CSP functions can be included in source code as a standard library via `#include <csp/*.h>`

**Testing performed**
1. Added `#include <csp/csp.h>` in `to_app.c`
1. Run `chmod +x run.sh` to make script executable
1. Run `./run.sh` to execute the script
1. Observe build output for errors - **none found**
1. Observed CFS log for errors - **none found**

**Expected behavior changes**
 - API Change: New API for CSP functions

**System(s) tested on**
 - OS: Ubuntu 18.04

**Additional context**
- Next steps will be incorporating the functionality in the CI/TO apps

**Contributor Info - All information REQUIRED for consideration of pull request**
Stephen Scott, McMaster NEUDOSE
